### DISCUSSION TITLE
Adding color scale

### DISCUSSION THREAD

**Discussion by skc1712** [Kitware/trame#76] [Q&A] [answered]
Hi,

I have just started with the examples given in the tutorial and was trying to understand if/how a 'color scale' can be added to the figure obtained from the file '/trame-tutorial/04_application/solution.py' to display the range values for each variable in the color map. It is possible to print the range value, but I am trying to have it displayed as 'color scale' alongside the figure for different color maps.

Also, in the 'slider' option, is it possible to display the minimum or maximum values for the slider? If so, it would be nice to know how to do that as well.

Thanks in advance,

Best,

**Comment by jourdain** [comment, marked as answer]
For the scalar bar you can follow [that example](https://kitware.github.io/vtk-examples/site/Cxx/Visualization/ScalarBarActor/) to set it on the server side.

Then for the range for the scalars for the slider, you can put them where ever you want in the UI. If you edit [that part](https://github.com/Kitware/trame-tutorial/blob/master/04_application/solution.py#L472-L494) with some dynamic text you will have those bounds

```python
    with ui_card(title="Contour", ui_name="contour") as content:
        vuetify.VSelect(
            # Contour By
            label="Contour by",
            v_model=("contour_by_array_idx", 0),
            items=("array_list", dataset_arrays),
            hide_details=True,
            dense=True,
            outlined=True,
            classes="pt-1",
        )
        content.add_child("{{ contour_min }}") # <--- NEW
        vuetify.VSlider(
            # Contour Value
            v_model=("contour_value", contour_value),
            min=("contour_min", default_min),
            max=("contour_max", default_max),
            step=("contour_step", 0.01 * (default_max - default_min)),
            label="Value",
            classes="my-1",
            hide_details=True,
            dense=True,
        )
        content.add_child("{{ contour_max }}") # <--- NEW
```

**Reply by skc1712** [reply]
Thank you. The range for the slider works; however, regarding the scalar bar, when I import the packages vtkScalarBarActor, vtkSphereSource and the others mentioned in the link, I get an import error. I do it in this way:

from vtkmodules.vtkRenderingCore import (
    vtkActor,
    vtkDataSetMapper,
    vtkRenderer,
    vtkRenderWindow,
    vtkRenderWindowInteractor,
    vtkSphereSource,        # <--- NEW
    vtkPolyData,            # <--- NEW
    vtkFloatArray,          # <--- NEW
    vtkScalarBarActor,      # <--- NEW

)

and get the error:

ImportError: cannot import name 'vtkSphereSource' from 'vtkmodules.vtkRenderingCore' (/home/sandekum/Desktop/INDOMINUS/dentalFEM/trame-tutorial/04_application/.venv/lib/python3.9/site-packages/vtkmodules/vtkRenderingCore.cpython-39-x86_64-linux-gnu.so)

Do you also know what could be the right way of doing it?

Thanks,

**Comment by jourdain** [comment]
The problem is that `vtkSphereSource`, ..., `vtkScalarBarActor` don't belong to the `vtkRenderingCore` module.
For that you need to find which module your VTK class belongs to. 

To do that:
1. [Find the class you want to import](https://github.com/Kitware/VTK/find/master)
2. Move to the up to the directory containing that class
3. Open `vtk.module` from that directory
4. Use the `LIBRARY_NAME` for importing that class

Example for vtkScalarBarActor
1. Start typing the name of the class 
![Screen Shot 2022-06-15 at 7 33 50 AM](https://user-images.githubusercontent.com/294518/173841342-775d5158-10fd-4395-829d-8c7ba2c576b2.png)
2. Go to parent directory
3. Open `vtk.module`
![Screen Shot 2022-06-15 at 7 35 02 AM](https://user-images.githubusercontent.com/294518/173840687-c7d01c9d-8948-417c-bccc-372e66897550.png)
4. Use vtkRenderingAnnotation 
  - `from vtkmodules.vtkRenderingAnnotation import vtkScalarBarActor`

**Reply by Datalong** [reply]
你好啊，我也是trame爱好者，最近在完善自己的剪辑功能部分py代码，发现 [#84](https://github.com/Kitware/trame/discussions/84)你写和我代码不谋而合，感觉还少点啥函数，希望学习下大神思路，请发到这个email,datalong_soft@foxmail.com,感谢你啦咱后面多讨论 @skumar1712
